master/buildbot/util/service.py

```diff
@@ -264,6 +268,7 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
                                     **config_sibling._config_kwargs)
 
     def setServiceParent(self, parent):
+        # XXX(sa2ajj): can we make 'master' a simple read-only property instead?
```

Comment:
Maybe I misunderstand the proposal -- AIUI the use of the property is only to prevent incorrectly setting `self.master` -- but we've never had that problem.  I don't think it makes sense to try to dynamically navigate back to the master through parent relationships on every request -- those parent relationships are not fixed.

What *would* be cool is some kind of "magic" master reference that we could get from a mixin, which somehow knows the current master.  That needs to be magic rather than just a global variable because we should be able to support multiple masters running in the same process (for example, for a graceful restart).  In synchronous code, that's pretty straightforward - a thread-local variable - but I'm not sure how to do it with async code.